src/loaders/mod_load.c

```diff
@@ -632,18 +677,25 @@ static int mod_load(struct module_data *m, HIO_HANDLE *f, const int start)
     /* Load and convert patterns */
     D_(D_INFO "Stored patterns: %d", mod->pat);
 
+    #ifndef LIBXMP_CORE_PLAYER
     if ((patbuf = (uint8 *) malloc(64 * 4 * mod->chn)) == NULL) {
 	return -1;
     }
+    #endif
 
     for (i = 0; i < mod->pat; i++) {
+	#ifndef LIBXMP_CORE_PLAYER
 	uint8 *mod_event;
+	#endif
 
 	if (libxmp_alloc_pattern_tracks(mod, i, 64) < 0) {
+	    #ifndef LIBXMP_CORE_PLAYER
 	    free(patbuf);
+	    #endif
 	    return -1;
 	}
 
+	#ifndef LIBXMP_CORE_PLAYER
```

Comment:
I think it would be better if -lite also used the buffer rather than one `hio_read` per event—this was an optimization we applied a while back to the main MOD loader but not to the -lite MOD loader for some reason. This *might* clean up these a bit, but I can also send a second patch after we merge this if you don't want to.

Reply:
Please send a patch for this